xla/service/gpu/transforms/cudnn_custom_call_compiler.cc

```diff
@@ -541,47 +246,134 @@ absl::StatusOr<se::gpu::CudnnGraph> HloCustomCallToCuDnnGraph(
     config.set_force_deterministic(force_deterministic);
     TF_RETURN_IF_ERROR(custom_call->set_backend_config(gpu_config));
 
```

Comment:
There's a lot of repetition here. How about putting some of these locals into a struct?

Reply:
Added TensorDescriptorFor() and MatmulTensorDescriptorFor() which remove lots of repeated code.